private-templates-service/client/src/store/currentTemplate/reducers.ts

```diff
@@ -0,0 +1,62 @@
+import { CurrentTemplateState, CurrentTemplateAction, REQUEST_NEW_TEMPLATE_UPDATE, FAILURE_NEW_TEMPLATE_UPDATE, RECEIVE_NEW_TEMPLATE_UPDATE, REQUEST_EXISTING_TEMPLATE_UPDATE, FAILURE_EXISTING_TEMPLATE_UPDATE, NEW_TEMPLATE, RECEIVE_EXISTING_TEMPLATE_UPDATE } from "./types";
+
+const initialState: CurrentTemplateState = {
+  templateID: undefined,
+  templateJSON: undefined,
+  templateName: undefined,
+  sampleDataJSON: undefined,
+  isFetching: false
+};
+
+export function currentTemplateReducer(state = initialState, action: CurrentTemplateAction): CurrentTemplateState {
+  switch (action.actionType) {
+    case NEW_TEMPLATE:
+      return {
+        ...state,
+        templateID: "",
+        templateJSON: JSON.stringify(require('../../assets/default-adaptivecards/defaultAdaptiveCard.json')),
+        templateName: "Untitled",
+        sampleDataJSON: "",
+        isFetching: false
+      };
+    case REQUEST_NEW_TEMPLATE_UPDATE:
+      return {
+        ...state,
+        isFetching: true
+      };
+    case FAILURE_NEW_TEMPLATE_UPDATE:
+      return {
+        ...state,
+        isFetching: false
```

Comment:
This one and the one above can be merged, no point having two reducers doing the same thing

Reply:
Also FAILURE_EXISTING_TEMPLATE_UPDATE

Reply:
Just wanted to confirm, should REQUEST_NEW_TEMPLATE_UPDATE and FAILURE_NEW_TEMPLATE_UPDATE be combined? By convention, REQUEST sets isFetching to true while FAILURE and RECEIVE set isFetching to false.

Reply:
If they are doing different stuff they can be seperate, if they are all setting it to true or something you can just merge them

Reply:
Looking at this again, I think he meant you could have had the action have a field `isFetching` and just assign `isFetching: action.isFetching`, then the two can be combined.

Reply:
^ what i suggested is optional. up to you

Reply:
I like Kody's suggestion